src/requirements/__test__/requirement-frontend-util.test.ts

```diff
@@ -127,6 +132,73 @@ it('computeFulfillmentCoursesAndStatistics course (without minNumberOfSlots) tes
   });
 });
 
+it('computeFulfillmentCoursesAndStatistics course (with additional requirements) test', () => {
+  const mockRequirement: RequirementWithIDSourceType = {
+    ...mockRequirementCommon,
+    fulfilledBy: 'courses',
+    perSlotMinCount: [2, 1, 1],
+    slotNames: ['', '', ''],
+    courses: [
+      [1, 2],
+      [3, 4],
+      [3, 4],
+    ],
+    additionalRequirements: {
+      R1: {
+        fulfilledBy: 'courses',
+        perSlotMinCount: [1, 1],
+        slotNames: ['', ''],
+        courses: [
+          [1, 3],
+          [2, 4],
+        ],
+      },
+      R2: {
+        fulfilledBy: 'courses',
+        perSlotMinCount: [1, 1],
+        slotNames: ['', ''],
+        courses: [
+          [1, 2],
+          [4, 3],
+        ],
+      },
+    },
+  };
+
+  // Asserts that the progress computation goes deeper into additional requirements,
+  // when they are present.
+  expect(
+    computeFulfillmentCoursesAndStatistics(
+      mockRequirement,
+      [getMockCourseTaken(1), getMockCourseTaken(2), getMockCourseTaken(3), getMockCourseTaken(4)],
+      {}
+    )
+  ).toEqual<StatisticsResultType>({
+    courses: [
+      [getMockCourseTaken(1), getMockCourseTaken(2)],
+      [getMockCourseTaken(3)],
+      [getMockCourseTaken(4)],
+    ],
+    fulfilledBy: 'courses',
+    minCountFulfilled: 4,
+    minCountRequired: 4,
+    additionalRequirements: {
+      R1: {
+        courses: [[getMockCourseTaken(1)], [getMockCourseTaken(2)]],
+        fulfilledBy: 'courses',
+        minCountFulfilled: 2,
+        minCountRequired: 2,
+      },
+      R2: {
+        courses: [[getMockCourseTaken(1)], [getMockCourseTaken(3)]],
+        fulfilledBy: 'courses',
+        minCountFulfilled: 2,
+        minCountRequired: 2,
+      },
+    },
+  });
+});
+
```

Comment:
Could we also test the situation where the base requirement is fulfilled by courses but R1 is fulfilled by credits (and R2 is fulfilled by courses)?